dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassReferenceGraph.java

```diff
@@ -50,38 +48,32 @@
   private final ImmutableSet<String> reachableClasses;
 
   static ClassReferenceGraph create(
-      Collection<SymbolReferenceSet> symbolReferenceSets, Set<Path> entryPointJars)
-      throws IOException {
-
-    // Collects all class symbol reference in symbolReferences
-    ImmutableSet<ClassSymbolReference> classSymbolReferences =
-        symbolReferenceSets.stream()
-            .flatMap(symbolReferenceSet -> symbolReferenceSet.getClassReferences().stream())
-            .collect(toImmutableSet());
+      SymbolReferenceMaps symbolReferenceMaps, Set<Path> entryPointJars) throws IOException {
 
     ImmutableSet.Builder<String> entryPointClassBuilder = ImmutableSet.builder();
     for (Path jar : entryPointJars) {
       for (String className : ClassDumper.listClassNamesInJar(jar)) {
         entryPointClassBuilder.add(className);
       }
     }
-
-    return new ClassReferenceGraph(classSymbolReferences, entryPointClassBuilder.build());
+    return new ClassReferenceGraph(
+        symbolReferenceMaps.getClassToClassSymbols(), entryPointClassBuilder.build());
   }
 
-  private ClassReferenceGraph(Set<ClassSymbolReference> classSymbolReferences,
+  private ClassReferenceGraph(
+      ImmutableSetMultimap<ClassFile, ClassSymbol> classSymbolReferences,
       Set<String> entryPointClasses) {
     MutableGraph<String> graph = GraphBuilder.directed().allowsSelfLoops(false).build();
 
-    for (ClassSymbolReference reference : classSymbolReferences) {
-      String sourceClassName = reference.getSourceClassName();
-      String targetClassName = reference.getTargetClassName();
-      if (sourceClassName.equals(targetClassName)) {
-        continue; // no self-loop
-      }
-      graph.putEdge(sourceClassName, targetClassName);
-    }
-
+    classSymbolReferences.forEach(
```

Comment:
What does this improve over the for loop? It feels like a gratuitous change.

Reply:
for-loop would require to use Map.Entry because we changed the class from `Set<ClassSymbolReference>` to `ImmutableSetMultimap<ClassFile, ClassSymbol>`

    for (Map.Entry<ClassFile, ClassSymbol> entry: classSymbolReferences.entries()) {
      String sourceClassName = entry.getKey().getClassName();
      String targetClassName = entry.getValue().getClassName();
      ...
    }

Thus `forEach` is more simpler.